### ISSUE TITLE
Error: No matching keys found when opening encrypted file

### ISSUE THREAD

**Issue by iensu** [anticomputer/age.el#8]
`rage` is not able to find the keys when opening a new file. I can create a new `.age` file and edit it just fine while the buffer is open, but if I kill the buffer and try to open it up again I get: 

```
run-hook-with-args-until-success: Opening input file: Age failed with error, No matching keys found
```

I've run `edebug` on `age--start` and can verify that the expected arguments are passed to `make-process`. I can run the command with the exact same arguments without problem in the terminal. Any idea what is going wrong?

**Environment info**
OS: MacOS 13.5
Emacs version: 29.1
Age program: rage 0.9.2
Age.el version number: 0.1.4

**Age.el configuration**

```lisp
(use-package age
  :ensure t
  :demand t
  :config
  (age-file-enable)
  (setq age-default-identity  "/Users/iensu/.ssh/id_ed25519"
        age-default-recipient "/Users/iensu/.ssh/id_ed25519.pub"
        age-pinentry-mode 'ask
        age-debug t
        age-program (executable-find "rage")))
```

**Comment by dywedir** [comment]
It seems that `rage` without a pinentry program is not supported by age.el (?)
Installing pinentry-gnome3 fixed a similar problem for me.


> https://github.com/str4d/rage#passphrases
If a binary named pinentry is available in $PATH, it will be used to ask the user for a passphrase. The PINENTRY_PROGRAM environment variable can be used to set the binary name or path to use. If a pinentry binary is not available, or PINENTRY_PROGRAM is set to the empty string, rage will fall back to the CLI instead.

**Comment by iensu** [comment]
Finally got around to fixing the issue. @dywedir, thanks to your comment I tried installing `pinentry-mac` and then it worked.  Setting the `PINENTRY_PROGRAM` environment variable did not work for me, instead I had to make sure the `pinentry` executable  referred to `pinentry-mac`, on my machine it was `pinentry-curses` and that didn't work. I created a symbolic link (`pinentry-mac` -> `pinentry`) to somewhere earlier in the `PATH`.
